src/ipc-handlers.ts

```diff
@@ -225,94 +221,64 @@ export async function createSite(
 		enableHttps,
 		siteId,
 		blueprint,
-		phpVersion = getWordPressProvider().DEFAULT_PHP_VERSION,
+		phpVersion,
+		noStart = false,
 	} = config;
 
-	const forceSetupSqlite = false;
-
 	const metric = getBlueprintMetric( blueprint?.slug );
 	bumpStat( StatsGroup.STUDIO_SITE_CREATE, metric );
 
-	// We only recursively create the directory if the user has not selected a
-	// path from the dialog (and thus they use the "default" or suggested path).
-	if ( ! ( await pathExists( path ) ) && path.startsWith( DEFAULT_SITE_PATH ) ) {
-		fs.mkdirSync( path, { recursive: true } );
-	}
-
-	if ( ! ( await isEmptyDir( path ) ) && ! isWordPressDirectory( path ) ) {
-		// Form validation should've prevented a non-empty directory from being selected
-		throw new Error( 'The selected directory is not empty nor an existing WordPress site.' );
-	}
-	let userData = await loadUserData();
-
-	const allPaths = userData?.sites?.map( ( site ) => site.path ) || [];
-	if ( allPaths.includes( path ) ) {
-		throw new Error( 'The selected directory is already in use.' );
-	}
-
-	const port = await portFinder.getOpenPort();
-
-	const details = {
-		id: siteId || crypto.randomUUID(),
-		name: siteName || nodePath.basename( path ),
-		path,
-		adminPassword: createPassword(),
-		port,
-		running: false,
-		phpVersion,
-		isWpAutoUpdating: wpVersion === getWordPressProvider().DEFAULT_WORDPRESS_VERSION,
-		customDomain,
-		enableHttps,
-	} as const;
+	try {
+		const { details: siteDetails } = await SiteServer.create(
+			{
+				path,
+				name: siteName,
+				wpVersion,
+				phpVersion,
+				customDomain,
+				enableHttps,
+				siteId,
+				blueprint: blueprint?.blueprint,
+				noStart,
+			},
+			{ wpVersion, blueprint: blueprint?.blueprint }
+		);
 
-	const server = SiteServer.create( details, { wpVersion, blueprint: blueprint?.blueprint } );
+		const parentWindow = BrowserWindow.fromWebContents( event.sender );
+		sendIpcEventToRendererWithWindow( parentWindow, 'theme-details-updating', {
+			id: siteDetails.id,
+		} );
```

Comment:
This method also needs the following calls:

```ts
				await server.updateCachedThumbnail();
				await sendThumbnailChangedEvent( event, id );
```

That logic doesn't depend on WP-CLI, so I believe we can add it in this PR